bigsea-controller/service/api/controller/plugins/pid/pid_alarm.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2017 LSD - UFCG.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from utils.logger import ScalingLog
+import datetime
+import time
+
+class PIDAlarm:
+
+    ERROR_METRIC_NAME = "application-progress.error"
+    
+    def __init__(self, actuator, metric_source, trigger_down, trigger_up, min_cap, max_cap, 
+                            metric_rounding, heuristic_options, application_id, instances):
+        # TODO: Check parameters
```

Comment:
Still **TODO**?

Reply:
Yes, I do not think it is going to be solved anytime soon.